src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,166 @@
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndexValue(index);
+        return foundElementByIndex(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndexValue(index);
+        Node<T> currentNode = foundElementByIndex(index);
+        T node = currentNode.item;
+        currentNode.item = value;
+        return node;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        T removedElement;
+        checkIndexValue(index);
+        if (index == 0) {
+            removedElement = head.item;
+            head = head.next;
+            if (head == null) {
+                tail = null;
+            }
+            size--;
+            return removedElement;
+        }
+        if (index == size - 1) {
+            removedElement = tail.item;
+            tail = tail.prev;
+            size--;
+            return removedElement;
+        } else {
+            Node<T> currentNode = foundElementByIndex(index);
+            removedElement = currentNode.item;
+            currentNode.prev.next = currentNode.next;
+            currentNode.next.prev = currentNode.prev;
+            size--;
+            return removedElement;
+        }
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> currentNode = head;
+        if (currentNode.item.equals(object)) {
+            head = currentNode.next;
+            size--;
+            return true;
+        }
+        for (int i = 0; i < size(); i++) {
+            if (currentNode.item == object || currentNode.item != null
+                    && currentNode.item.equals(object)) {
+                currentNode.prev.next = currentNode.next;
+                if (currentNode.next != null) {
+                    currentNode.next.prev = currentNode.prev;
+                } else {
+                    tail = currentNode.prev;
+                    tail.next = null;
+                }
+                size--;
+                return true;
+            }
+            currentNode = currentNode.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private class Node<T> {
+        private T item;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(T item) {
+            this.item = item;
+        }
+    }
+
+    private void checkIndexValue(int index) {
+        if (index >= size || index < 0) {
+            throw new IndexOutOfBoundsException("Index " + index
+                    + " must be less than " + size());
+        }
+    }
+
+    private Node<T> foundElementByIndex(int index) {
+        Node<T> node = head;
+        for (int i = 0; i < index; i++) {
+            node = node.next;
+        }
```

Comment:
worst case scenario is to iterate N nodes where N is size of the list
rewrite your solution to make worst case N/2
[checklist issue](https://github.com/mate-academy/jv-linked-list/blob/master/checklist.md#dont-iterate-your-nodes-from-the-head-to-the-tail-or-vice-versa)